src/otx/core/data/dataset/base.py

```diff
@@ -86,6 +90,7 @@ def __init__(
         stack_images: bool = True,
         to_tv_image: bool = True,
         data_format: str = "",
+        collate_mode: Literal["torch", "numpy"] = "torch",
```

Comment:
I am in favour of using enumerations as strings are quite susceptible to typos thus introducing errors. We can also just type this as 
```python
collate_mode: CollateMode | str = CollateMode.Torch,
```

And below,
```python
self.collate_mode = CollateMode(collate_mode)
```